File: Swift/V3/SfPoint.swift
//
//  SfPoint.swift
//  Surface Swift
//
//  Created by Webchimp on 26/09/16.
//  Copyright © 2016 AppBuilders. All rights reserved.
//

import Foundation

class SfPoint: NSObject {
    
    var x: Float! = 0.0;
    var y:Float! = 0.0;
    
    func setPoint(_ x:Float, y:Float) -> Void {
        
        self.x = x;
        self.y = y;
    }
    
    func isEmpty() -> Bool {
        
        if ( self.x == nil && self.y == nil ) {
            return true;
        }
        return false;
    }
    
    

}

File: Java/SurfaceDemo/surface/src/main/java/com/appbuilders/surface/SfPoint.java
package com.appbuilders.surface;

/**
 * Created by ercsanchez on 26/09/17.
 */

public class SfPoint extends Object {

    public float x;
    public float y;

    public SfPoint() {
        this.x = 0;
        this.y = 0;
    }

    public SfPoint(float x ,float y) {
        this.setPoint(x, y);
    }

    public void setPoint(float x ,float y) {
        this.x = x;
        this.y = y;
    }

    public boolean isEmpty() {
        return (this.x == 0 && this.y == 0);
    }
}
